lib/dotenv/parser.rb

```diff
@@ -59,8 +59,6 @@ def parse_line(line)
         if variable_not_set?(line)
           raise FormatError, "Line #{line.inspect} has an unset variable"
         end
-      elsif line !~ /\A\s*(?:#.*)?\z/ # not comment or blank line
-        raise FormatError, "Line #{line.inspect} doesn't match format"
```

Comment:
I don't have a strong feeling about it, but I would think not. We don't expect every line to follow this format, so we shouldn't raise a warning.